### DISCUSSION TITLE
Why are custom errors not de-duplicated across strategies?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [allo-protocol/allo-v2#138] [Q&A]
Many custom errors, e.g. `UNAUTHORIZED`, are duplicated across multiple strategy contracts:

- https://github.com/allo-protocol/allo-v2/blob/04f84f89c073e4fd761b6864cf198df994cda02f/contracts/strategies/direct-grants-simple/DirectGrantsSimpleStrategy.sol#L48
- https://github.com/allo-protocol/allo-v2/blob/04f84f89c073e4fd761b6864cf198df994cda02f/contracts/strategies/donation-voting/DonationVotingStrategy.sol#L46

Why are they not de-duplicated by defining them only in [`BaseStrategy`](https://github.com/allo-protocol/allo-v2/blob/04f84f89c073e4fd761b6864cf198df994cda02f/contracts/strategies/BaseStrategy.sol)?

**Comment by PaulRBerg** [comment]
Alternatively, an [errors library](https://twitter.com/PaulRBerg/status/1587041303015165952) design could have been employed.

An example can be found in Sablier's [V2 Core](https://github.com/sablier-labs/v2-core/blob/fcbc567aea6362c1f102eb19732badeb4899dec5/src/libraries/Errors.sol) repository.

**Reply by codenamejason** [reply]
This is great feedback and honestly would love to put it on the todos.
